lib/msf/base/sessions/meterpreter.rb

```diff
@@ -302,11 +302,18 @@ def desc
   ##
   # :category: Msf::Session::Scriptable implementors
   #
-  # Runs the meterpreter script in the context of a script container
+  # Runs the Meterpreter script or resource file
   #
   def execute_file(full_path, args)
-    o = Rex::Script::Meterpreter.new(self, full_path)
-    o.run(args)
+    # Infer a Meterpreter script by it either having the .rb extension, or it
+    # containing a reference to the client object. This is for backward
+    # compatibility, since the API is not explicit to the user whether this
+    # should be a resource file or a Meterpreter script.
+    if File.extname(full_path) == ".rb" || File.read(full_path).match?(/\s*client\/./)
```

Comment:
Is the regex is to catch a Meterpreter script that doesn't end in `.rb`? What if the user has a resource script that contains a line that matches the regex, for example, in a comment?

Reply:
The regex looks for any line that _begins_ with `client.`, so it would not catch a comment. I could just have it only base the decision on file extension alone if desired. Every meterpreter script I know of does a 'client.' dereference though, otherwise it would not be very useful.

Reply:
Basically it says a meterpreter script must either end in .rb (which every one I know of does), and at some point use the client object (also true of every meterpreter script), otherwise it is interpreted as a resource script.